src/addrman.h

```diff
@@ -99,8 +99,14 @@ class AddrMan
     template <typename Stream>
     void Unserialize(Stream& s_);
 
-    //! Return the number of (unique) addresses in all tables.
-    size_t size() const;
+    /**
+    * Return size information about addrman.
+    *
+    * @param[in] net              Select addresses only from specified network (nullopt = all)
+    * @param[in] in_new           Select addresses only from one table (true = new, false = tried, nullopt = both)
+    * @return                     Number of unique addresses that match specified options.
+    */
+    size_t Size(std::optional<Network> net = {}, std::optional<bool> in_new = {}) const;
```

Comment:
nit, if you re-touch: usually `std::nullopt` is used instead of `{}` and I think it is more readable, e.g.:

https://github.com/bitcoin/bitcoin/blob/79e18ebc81f112f989ff50fb6f5a6a17db9b6758/src/streams.h#L236

Reply:
will do if I need to re-touch for another reason.

Reply:
done in #27015